### PR TITLE
Enable Strict Template

### PR DESCRIPTION
**Pull request by iozbeyli**
:rotating_light:**ENABLES STRICT TEMPLATE**:rotating_light:

Fix strict template errors in:

exam-navigation-bar.component.ts
programming-exam-submission.component.html
modeling-exam-summary.component.html
quiz-exam-summary.component.html
modeling-exercise.component.html
modeling-submission.component.html
code-editor-container.component.html
drag-item.component.html
exercise-assessment-dashboard.component.html
header-exercise-page-with-details.component.ts
textblock-assessment-card.component.html
markdown-editor.component.html
settings.component.html
assessment-complaint-alert.component.html
assessment-layout.component.html
course-card.component.ts
exercise-unit.component.html
course-statistics.component.ts
participant-scores.service.ts

### CODE REVIEW

File: src/main/webapp/app/overview/course-card.component.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { ExerciseService } from 'app/exercises/shared/exercise/exercise.service'
 import { CourseStatisticsDataSet } from 'app/overview/course-statistics/course-statistics.component';
 import { CachingStrategy } from 'app/shared/image/secured-image.component';
 import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import { Color } from 'ng2-charts';
```
Comment (krusche) [inline src/main/webapp/app/overview/course-card.component.ts:11]: please remove this line

File: src/main/webapp/app/shared/participant-scores/participant-scores.service.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { HttpClient, HttpParams, HttpResponse } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { SERVER_API_URL } from 'app/app.constants';
 import { Observable } from 'rxjs';
+import { BaseEntity } from '../model/base-entity';
```
Comment (krusche) [inline src/main/webapp/app/shared/participant-scores/participant-scores.service.ts:5]: please remove

File: tsconfig.json [hunk 3]
```diff
@@ -31,7 +31,7 @@
         "genDir": "build/resources/main/aot",
         "skipMetadataEmit" : true,
         "fullTemplateTypeCheck": true,
-        "strictTemplates": false,
+        "strictTemplates": true,
```
Comment (krusche) [inline tsconfig.json:34]: 👍

File: src/main/webapp/app/shared/participant-scores/participant-scores.service.ts (hunk 2 above)
Comment (iozbeyli) [reply src/main/webapp/app/shared/participant-scores/participant-scores.service.ts]: Removed it

File: src/main/webapp/app/overview/course-card.component.ts (hunk 1 above)
Comment (iozbeyli) [reply src/main/webapp/app/overview/course-card.component.ts]: Removed it

**Review by SimonEntholzer — APPROVED** [review]
Tested on TS2. Changes look good to me and did not break anything

**Review by ivohashamov — APPROVED** [review]
Code changes make sense, and I couldn't find anything broken on TS2

**Review by PatZintel — APPROVED** [review]
tested on ts1, clicked through the various UI-elements and nothing out of the ordinary

**Comment by coolchock** [comment]
tested on ts1, everything works fine

**Review by coolchock — COMMENTED** [review]
tested on ts1, everything works fine

**Pull request merged**